trio/tests/test_path.py

```diff
@@ -119,3 +133,14 @@ class MockWrapper:
     Type.generate_wraps(MockWrapper, {'unsupported': None})
 
     assert not hasattr(MockWrapper, '_private')
+
+
+async def test_path_wraps_path(path):
+    other = trio.Path(path)
+
+    assert path == other
```

Comment:
Testing that this succeeds is good, but we should also have a test that `trio.Path(1)` or `trio.Path(object())` fails :-)